MetaMask/metamask-extension
# update send slice to work with EIP 1559 networks

Opening for QA

### What does this PR do? 
1. updates the send slice state to work with either gasPrice or maxFeePerGas/maxPriorityFeePerGas, including generating a fully formed type 2 (0x2) EIP 1559 transaction for submission.
2. Hooks up the minimum gas paid estimation on confirmations to work with EIP 1559 fields.

### Steps to QA:
With SHOW_EIP_1559_UI set to `0` or `false`:
1. Try to send transactions on the major test nets and make sure that the confirmation page quotes align with expectations.
2. Do this on mainnet, but on testnets also submit transactions and confirm the fields in the block explorer.
3. Modify gas prices using the edit modal and make sure the changes are reflected in the confirmation screen AND on the block explorer on test nets.

With SHOW_EIP_1559_UI set to `1`:
1. Same as above, but note you will not see the gas fields on send screen so it becomes more difficult to validate. You can check redux but I have had some trouble with it since the build system was updated.
2. You'll want to test sending transactions on the various testnets and mainnet. Note, mainnet will be a gasPrice network. You can also run ganache locally and connect to local host for a test network to test sending gasPrice transactions on.
3. In confirmation screens the gas estimate will be `estimatedBaseFee + maxPriorityFeePerGas` or maxFeePerGas, whichever is lowest.
